src/core/qgsnewsfeedparser.h

```diff
@@ -0,0 +1,169 @@
+/***************************************************************************
+    qgsnewsfeedparser.h
+    -------------------
+    begin                : July 2019
+    copyright            : (C) 2019 by Nyall Dawson
+    email                : nyall dot dawson at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef QGSNEWSFEEDPARSER_H
+#define QGSNEWSFEEDPARSER_H
+
+#include "qgis_core.h"
+#include "qgis_sip.h"
+#include <QObject>
+#include <QUrl>
+#include <QPixmap>
+
+class QgsNetworkContentFetcher;
+
+/**
+ * \ingroup core
+ * Parser for published QGIS news feeds.
+ *
+ * This class is designed to work with the specialized QGIS news feed API. See
+ * https://github.com/elpaso/qgis-feed.
+ *
+ * \since QGIS 3.10
+ */
+class CORE_EXPORT QgsNewsFeedParser : public QObject
+{
+    Q_OBJECT
+  public:
+
+    /**
+     * Represents a single entry from a news feed.
+     */
+    class Entry
```

Comment:
/nitpickmode
Data only classes are better made structs
/nitpickmode off

Reply:
We lose Python class docs if I change this to a struct though...

Reply:
Seriously? Ouch!